src/Psy/Configuration.php

```diff
@@ -140,17 +147,34 @@ public function getConfigFile()
         foreach ($this->getConfigDirs() as $dir) {
             $file = $dir . '/config.php';
             if (@is_file($file)) {
-                return $this->configFile = $file;
+                return $file;
             }
 
             $file = $dir . '/rc.php';
             if (@is_file($file)) {
-                return $this->configFile = $file;
+                return $file;
```

Comment:
@bobthecow this seemed like the cleanest way to ensure that checking `$this->configFile` would only be set by the user providing a config file. It does have a (small) performance implication if the user calls `getConfigFile` more than once. By default there is only one invocation, so it seemed safe to change.